### PR TITLE
Y2K-1297 remove multi users rpc

### PR DESCRIPTION
**Pull request by Daniel-Wang**

### CODE REVIEW

File: protocol/avdl/keybase1/teams.avdl [hunk 1]
```diff
@@ -1233,6 +1233,19 @@ protocol teams {
   // Note: allowInaction only applies to invite removals, just because that was what was needed.
   void teamRemoveMember(int sessionID, TeamID teamID, string username, string email, TeamInviteID inviteID, boolean allowInaction);
 
+  record RemoveMemberArgs {
```
Comment (Daniel-Wang) [inline protocol/avdl/keybase1/teams.avdl]: @mlsteele don't know if there's a better name for this

File: go/teams/member_test.go [hunk 2]
```diff
@@ -449,6 +449,43 @@ func TestMemberRemoveWithoutDemotion(t *testing.T) {
...
+
+	rolePairB := keybase1.RemoveMemberArgs{
+		Username: otherB.Username,
+	}
+
+	users := []keybase1.RemoveMemberArgs{rolePairA, rolePairB}
+
+	res, err := RemoveMembers(context.TODO(), tc.G, teamID, users)
+	require.NoError(t, err)
+	require.Empty(t, res.Failures)
```
Comment (jzila) [inline go/teams/member_test.go]: Please assert `TeamRole_NONE` for both users after the removal.

File: go/teams/service_helper.go [hunk 3]
```diff
@@ -987,6 +987,58 @@ func MemberRoleFromID(ctx context.Context, g *libkb.GlobalContext, teamID keybas
...
+		if len(user.Username) > 0 {
+			exclusiveActions = append(exclusiveActions, "username")
+		}
+		if len(user.Email) > 0 {
+			exclusiveActions = append(exclusiveActions, "email")
+		}
+		if len(user.InviteID) > 0 {
+			exclusiveActions = append(exclusiveActions, "inviteID")
+		}
+		if len(exclusiveActions) > 1 {
```
Comment (jzila) [inline go/teams/service_helper.go]: If more than one of them is set, then the length of the concatenation of the actions will be greater than the length of any one of them. Then you don't need a slice.
```go
keysLen := len(user.Username + user.Email + string(user.InviteID))
if keysLen > len(user.Username) && keysLen > len(user.Email) && keysLen > len(user.InviteID) {
```

File: protocol/avdl/keybase1/teams.avdl (hunk 1 above)
Comment (jzila) [reply protocol/avdl/keybase1/teams.avdl]: `TeamMemberToRemove`

File: go/teams/service_helper.go (hunk 3 above)
Comment (Daniel-Wang) [reply go/teams/service_helper.go]: Though this way may lose the nice formatting of the slice for the log (with spaces or brackets). I just copied the logic from what was in the remove-single-member code.

Comment (Daniel-Wang) [reply go/teams/service_helper.go]: The best way i thought with the suggested approach was something like:
```
exclusiveActions := user.Username + user.Email + string(user.InviteID)
		lenExclActions := len(exclusiveActions)
		
		if lenExclActions > len(user.Username) && lenExclActions > len(user.Email) && lenExclActions > len(user.InviteID) {
			g.Log.CDebugf(ctx, "RemoveMembers: can only do 1 of [%s, %s, %s] at a time", user.Username, user.Email, user.InviteID)
			failedToRemove = append(failedToRemove, user)
			continue
		}
```

Comment (jzila) [reply go/teams/service_helper.go]: Sure, same thing.

**Pull request merged**